_posts/2015-11-13-svg-sprites-angular.md

```diff
@@ -0,0 +1,167 @@
+---
+layout: post
+title: Svg sprites
+permalink: /svg-sprites
+tags: angular, svg, sprite
+author_id: sinisag
+category: webapp
+---
+
+(CSS) Sprite is a method, where we combine multiple small images or graphic in to one large image and access the needed graphic by specifying 'coordinates' on the big image.
+The main reason behind it is to improve performance of website. To understand this better, we need to be aware of browsers technical limitations.
+
+## Browser limitations
+
+- Limited amount of concurrent connections ([more] (http://www.browserscope.org/?category=network&v=top))
+- Limited amount of concurrent connections to the same hostname
+
+The second limitation we can tackle by serving assets from different hosts, but we are still limited with amount of concurrent connections.
+
+In addition every http request comes with certain amount of overhead. We can partly address this by serving assets from cookie free domains (which should be the case regardless if you use sprites or not), but we can't fully prevent the overhead.
+
+So generating sprites became very popular way to overcome those limitations. With joining multiple icons/assets in to one big asset sheet, we download all related assets with one http request.
+Of course, there is some discussion about what makes sense to group in sprites, to download only needed assets and not something used rarely, but that discussion is out of the scope of this article.
+
+Ok, using sprites is nice but why svg sprites?
+
+Imagine, you have a set of icons that you group in sprite and you have to support multiple devices (as we usually do) - desktop, ios devices, mobile devices, tablets,... - with different screens. In that case, you can't use the same sprite if you wan to utilize screen specific advantages. For example, retina screens need assets double the size then other screens and serving bigger icons to screens that do not need them,
+would cause an overhead. That is bad for multiple reasons, most obvious one being downloading more data then needed and therefore increase of download time.
+
+Ok..so I create 2 sprite maps and serve them based on device type? Well, is not that simple, what if your icons change color based on context they are displayed in? Very quickly, you can end up with numerous amount of sprite files and complex logic to deal with them. At this point things can get out of hand very quickly.
+
+## There is a better way!
+
+Meet [svgs] (https://sl.wikipedia.org/wiki/Scalable_Vector_Graphics)! Svg stands for Scalable Vector Graphic. Main advantage of svgs is that they are, as name suggests, scalable vector graphics. That means they will be sharp on all type of screens and in all sizes - you need just one asset to achieve that. But it gets better, you can also change icon color, width, height (and much more!) through css properties. That means, that if you can have your repeatable icons in form of svgs, then you have to make only one sprite and it will serve any purpose.
```

Comment:
I think the English version of the page would be clearer :D https://en.wikipedia.org/wiki/Scalable_Vector_Graphics